tinygrad/shape/symbolic.py

```diff
@@ -51,12 +51,12 @@ def __rmul__(self, b:int): return self*b
   def __rfloordiv__(self, b:int): return NumNode(b) // self
   def __floordiv__(self, b:Union[Node,int], factoring_allowed=True):
     if isinstance(b, Node):
-      if b.__class__ is NumNode: return self // b.b
+      if b.__class__ is NumNode: return self.__floordiv__(b.b, factoring_allowed)
```

Comment:
the problem was something like `Node.__floordiv__(NumNode(-16), factoring_allowed=False)` was being called, this ran and called itself but didn't give factoring_allowed so it ended up being `Node.__floordiv__(-16, factoring_allowed=True)`

Reply:
Can you add a direct test for this case in test/unit/test_symbolic.py?

Reply:
yup

Reply:
test is added now, you can see the subtleness of the bug

Reply:
thanks. yea that looks very sketchy, we probably have other bugs around div negative numbers. I will think about how to improve that.

change looks good. I will run more tests tomorrow and merge the fix if all looks good!